test/hummingbot/strategy/cross_exchange_market_making/test_cross_exchange_market_making_gateway.py

```diff
@@ -0,0 +1,1017 @@
+                    limit_order.quantity,
+                    AddedToCostTradeFee(Decimal(0)),
+                ),
+            )
+            market.trigger_event(
+                MarketEvent.BuyOrderCompleted,
+                BuyOrderCompletedEvent(
+                    market.current_timestamp,
+                    limit_order.client_order_id,
+                    base_currency,
+                    quote_currency,
+                    base_currency_traded,
+                    quote_currency_traded,
+                    OrderType.LIMIT,
+                ),
+            )
+        else:
+            market.set_balance(quote_currency, market.get_balance(quote_currency) + quote_currency_traded)
+            market.set_balance(base_currency, market.get_balance(base_currency) - base_currency_traded)
+            market.trigger_event(
+                MarketEvent.BuyOrderCreated,
+                SellOrderCreatedEvent(
+                    market.current_timestamp,
+                    OrderType.LIMIT,
+                    limit_order.trading_pair,
+                    limit_order.quantity,
+                    limit_order.price,
+                    limit_order.client_order_id,
+                    limit_order.creation_timestamp * 1e-6,
+                )
+            )
+            market.trigger_event(
+                MarketEvent.OrderFilled,
+                OrderFilledEvent(
+                    market.current_timestamp,
+                    limit_order.client_order_id,
+                    limit_order.trading_pair,
+                    TradeType.SELL,
+                    OrderType.LIMIT,
+                    limit_order.price,
+                    limit_order.quantity,
+                    AddedToCostTradeFee(Decimal(0)),
+                ),
+            )
+            market.trigger_event(
+                MarketEvent.SellOrderCompleted,
+                SellOrderCompletedEvent(
+                    market.current_timestamp,
+                    limit_order.client_order_id,
+                    base_currency,
+                    quote_currency,
+                    base_currency_traded,
+                    quote_currency_traded,
+                    OrderType.LIMIT,
+                ),
+            )
+
+    @staticmethod
+    def emit_order_created_event(market: Union[MockPaperExchange, MockAMM], order: LimitOrder):
+        event_cls = BuyOrderCreatedEvent if order.is_buy else SellOrderCreatedEvent
+        event_tag = MarketEvent.BuyOrderCreated if order.is_buy else MarketEvent.SellOrderCreated
+        market.trigger_event(
+            event_tag,
+            message=event_cls(
+                order.creation_timestamp,
+                OrderType.LIMIT,
+                order.trading_pair,
+                order.quantity,
+                order.price,
+                order.client_order_id,
+                order.creation_timestamp * 1e-6
+            )
+        )
+
+    @async_test(loop=ev_loop)
```

Comment:
Same comment here for @async_test as before

Reply:
ditto

Reply:
✅